packages/kit/src/core/sync/write_types/test/actions/+page.server.js

```diff
@@ -35,3 +35,57 @@ export const actions = {
 		return fail(400);
 	}
 };
+
+/**
+ * Ordinarily this would live in a +page.svelte, but to make it easy to run the tests, we put it here.
```

Comment:
I still don't really get what this does. How is it invoked? Or is there something that's altered by the presence of this function?

Is having a `submit` function in a `+page.server.js` a feature of the framework? (if so, I can't find it in the docs)

Reply:
No, it would go in a `+page.svelte` (as the comment says 😉 )

```svelte
<script lang="ts">
  import { enhance } from '$app/forms';
  import type { SubmitFunction } from './$types';

  const submit: SubmitFunction = () => {
    // ...
  };
</script>

<form method="POST" use:enhance={submit}>
```

Reply:
What's it doing here? It's not tied to a `use:enhance`, so I'm not sure how it's getting used or what impact it's having

Reply:
It's verifying that the types are being generated correctly. Putting the `submit` function here instead of in a component saves us from having to build an entire project just to unit test the type generation